src/core/WakaTimeCore.ts

```diff
@@ -38,13 +48,203 @@ class WakaTimeCore {
+      const cleanLine = lines[i]
+        .trim()
+        .replace(/(\/|@@)$/, '')
+        .replace(/^(?:https?:\/\/)?/i, '');
+      if (cleanLine === '') continue;
+
+      const projectIndicatorIndex = cleanLine.lastIndexOf(projectIndicatorCharacters);
+      const projectIndicatorExists = projectIndicatorIndex > -1;
+      let projectName = null;
+      let urlFromLine = cleanLine;
+      if (projectIndicatorExists) {
+        const start = projectIndicatorIndex + projectIndicatorCharacters.length;
+        projectName = cleanLine.substring(start);
+        urlFromLine = cleanLine
+          .replace(cleanLine.substring(projectIndicatorIndex), '')
+          .replace(/\/$/, '');
+      }
+      const schemaHttpExists = url.match(/^http:\/\//i);
+      const schemaHttpsExists = url.match(/^https:\/\//i);
+      let schema = '';
+      if (schemaHttpExists) {
+        schema = 'http://';
+      }
+      if (schemaHttpsExists) {
+        schema = 'https://';
+      }
+      const cleanUrl = url
+        .trim()
+        .replace(/(\/|@@)$/, '')
+        .replace(/^(?:https?:\/\/)?/i, '');
+      const startsWithUrl = cleanUrl.toLowerCase().includes(urlFromLine.toLowerCase());
+      if (startsWithUrl) {
+        return {
+          project: projectName,
+          url: schema + urlFromLine,
+        };
+      }
+    }
+
+    return {
+      project: null,
+      url: null,
+    };
+  }
+
+  /**
+   * Given the heartbeat and logging type it creates a payload and
+   * sends an ajax post request to the API.
+   *
+   * @param heartbeat
+   * @param debug
+   */
+  async sendHeartbeat(heartbeat: SendHeartbeat, apiKey: string): Promise<void> {
+    let payload;
+
+    const loggingType = await this.getLoggingType();
+    // Get only the domain from the entity.
+    // And send that in heartbeat
+    if (loggingType == 'domain') {
+      heartbeat.url = getDomainFromUrl(heartbeat.url);
+      payload = this.preparePayload(heartbeat, 'domain');
+      await this.sendPostRequestToApi(payload, apiKey);
+    }
+    // Send entity in heartbeat
+    else if (loggingType == 'url') {
+      payload = this.preparePayload(heartbeat, 'url');
+      await this.sendPostRequestToApi(payload, apiKey);
+    }
+  }
+
+  /**
+   * Returns a promise with logging type variable.
+   *
+   * @returns {*}
+   * @private
+   */
+  async getLoggingType(): Promise<string> {
+    const items = await browser.storage.sync.get({
+      loggingType: config.loggingType,
+    });
+
+    return items.loggingType;
+  }
+
+  /**
+   * Creates payload for the heartbeat and returns it as JSON.
+   *
+   * @param heartbeat
+   * @param type
+   * @param debug
+   * @returns {*}
+   * @private
+   */
+  preparePayload(heartbeat: SendHeartbeat, type: string): Record<string, unknown> {
+    const payload: Record<string, unknown> = {
+      entity: heartbeat.url,
+      time: moment().format('X'),
+      type: type,
+      user_agent: 'browser-wakatime/' + config.version,
```

Comment:
Is it possible to set either browser's version and the extension version?

Reply:
Can you please provide an example?

will be something like this? `browser-wakatime/Chrome/109.0.0.0`

Reply:
It should be something like `chrome/<chrome_version> chrome-wakatime/<extension_version>`. @alanhamlett would you confirm this?

Example: `vscode/1.74.2 vscode-wakatime/22.1.0`

Reply:
@gandarez 
I changed userAgent to se like this `"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36 browser-wakatime/3.0.0"`
please let me know what you think about it

Reply:
from here https://github.com/wakatime/chrome-wakatime/pull/160/commits/a4baf7cf8600bc9c637a568f5c12208987a8f8d4
![image](https://user-images.githubusercontent.com/1050904/215163722-de21f50c-2c4b-4f5b-b540-8ae385c8b52d.png)